src/charm.py

```diff
@@ -4,39 +4,89 @@
 """A Juju charm for OpenTelemetry Collector on Kubernetes."""
 
 import os
-from typing import Any, Dict
+from typing import Any, Dict, List
+from pathlib import Path
+import yaml
+import logging
 
-from config import Config
+from config import Config, Ports
 
 from ops import CharmBase, main
 from ops.model import ActiveStatus
 from ops.pebble import Layer
+from charms.prometheus_k8s.v0.prometheus_scrape import (
+    MetricsEndpointConsumer,
+)
+from charms.prometheus_k8s.v1.prometheus_remote_write import (
+    PrometheusRemoteWriteConsumer,
+)
+from collections import namedtuple
+import shutil
+from cosl import JujuTopology
+
+logger = logging.getLogger(__name__)
+RulesMapping = namedtuple("RulesMapping", ["src", "dest"])
+
+
+def _aggregate_alerts(alerts: Dict, rule_path_map: RulesMapping):
+    if os.path.exists(rule_path_map.dest):
+        shutil.rmtree(rule_path_map.dest)
+    shutil.copytree(rule_path_map.src, rule_path_map.dest)
+    for topology_identifier, rule in alerts.items():
+        rule_file = Path(rule_path_map.dest) / f"juju_{topology_identifier}.rules"
+        rule_file.write_text(yaml.safe_dump(rule))
+        logger.debug(f"updated alert rules file {rule_file.as_posix()}")
 
 
 class OpenTelemetryCollectorK8sCharm(CharmBase):
     """Charm to run OpenTelemetry Collector on Kubernetes."""
 
     _config_path = "/etc/otelcol/config.yaml"
     _container_name = "otelcol"
+    _rules_source_path = "src/prometheus_alert_rules"
+    _rules_destination_path = "prometheus_alert_rules"
 
     def __init__(self, *args):
         super().__init__(*args)
-        if not self.unit.get_container("otelcol").can_connect():
+        if not self.unit.get_container(self._container_name).can_connect():
             return
```

Comment:
Should we set status before return? E.g., in prom we have:
- `MaintenanceStatus("Waiting for prometheus to start")`
- `MaintenanceStatus("Configuring Prometheus")`